crates/core/tedge_mapper/src/c8y/tests.rs

```diff
@@ -49,6 +49,22 @@ async fn mapper_publishes_a_supported_operation_and_a_pending_operations_onto_c8
     // Start SM Mapper
     let sm_mapper = start_sm_mapper(broker.port).await;
 
+    let mut msg = messages
+        .recv()
+        .with_timeout(TEST_TIMEOUT_MS)
+        .await
+        .expect_or("No message received before timeout");
+    dbg!(&msg);
+
+    while !msg.contains("114") {
```

Comment:
Is this logic correct? Do you really want to keep reading messages until that `114` message is received? There's not guarantee that this `114` message would be appearing if the test is run on a machine where `tedge` is installed before running this integartion test (often the case in our workflow VMs which are clean, without tedge installations, while running cargo tests).

I assume what you wanted is to ignore the `114` message even if it's seen on `c8y/s/us` and only process the subsequent messages.
```suggestion
    if msg.starts_with("114") {
```

Reply:
Needs to skip messages until the tested messages are expected, 114 and 118 may look different on different test environments.